_posts/en/newsletters/2020-07-01-newsletter.md

```diff
@@ -0,0 +1,355 @@
+  about the progress of restoring from a backup.
+
+- [BIPs #933][] adds the [BIP339][] specification for transaction
+  relay announcement using Witness Transaction Identifiers (wtxids).
+  Currently, nodes announce new unconfirmed transactions to their peers
+  using the transaction's txid, which is a hash over the legacy fields
+  of the transaction.  The new fields used in segwit transactions are
+  not included in the txid, which was necessary to eliminate third-party
+  and counterparty transaction malleability.  However, during the review
+  of segwit, Peter Todd [noticed][todd relay malleability] that
+  announcing transactions by their txid could allow nodes to modify the
+  segwit fields before relaying a transaction.  This couldn't be used to
+  steal money directly, but it did allow a relay node to mutate a valid
+  transaction into an invalid or unacceptable transaction without
+  changing its txid.
+
+    Before Todd's discovery, this was a problem: Bitcoin Core would
+    track the txids of invalid transactions it had recently seen and
+    refuse to download them again.  That meant a malicious node could
+    prevent any of its peers from downloading a valid segwit transaction
+    from any of their other peers by sending them a mutated version of
+    that transaction.  Arbitrary transaction censorship, even just at
+    the relay level, is bad by itself---but it has especially severe
+    consequences for time-sensitive protocols such as LN.
+
+    At the time of Todd's analysis, segwit was nearing release, so a
+    [quick fix][Bitcoin Core #8312] was implemented that prevents
+    Bitcoin Core from caching the rejection status of segwit
+    transactions that fail for the type of witness-related errors that
+    malicious nodes can inject.  This eliminated the issue, but it means
+    that Bitcoin Core uses more bandwidth than necessary when it
+    encounters segwit transactions that are invalid because of
+    accidental mistakes.  It may also complicate the development of new
+    relay protocols such as [package relay][topic package relay].
+
+    The long-term solution to the problem is that transactions should be
+    announced using a hash that commits to the entire transaction---both
+    the legacy fields and the new segwit fields.  That's exactly what
+    wtxids do, and they're already needed in the protocol to construct
+    the witness merkle root in each block's coinbase transaction.  This
+    new BIP proposes updating the P2P protocol's `inv` message that
+    announces new transactions, and the `getdata` message that requests
+    a transaction, to allow them both to use wtxids.  That will allow
+    nodes to skip re-downloading a transaction if it has the same wtxid
+    as a transaction that was previously found to be invalid or
+    unacceptable.
+
+    The BIP339 proposal also adds the additional feature negotiation
+    between newly started nodes described in [Newsletter #87][news87
+    negotiation].
```

Comment:
Is there value in mentioning the [implementation](http://github.com/bitcoin/bitcoin/pull/18044)? (I'm aware you can also get to it from the BIP)